url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/util/mxUtils.java

```diff
@@ -0,0 +1,2511 @@
+				: Font.PLAIN;
+
+		return new Font(fontFamily, swingFontStyle, (int) (fontSize * scale));
+	}
+
+	/**
+	 * 
+	 */
+	public static String hexString(Color color)
+	{
+		int r = color.getRed();
+		int g = color.getGreen();
+		int b = color.getBlue();
+
+		return String.format("#%02X%02X%02X", r, g, b);
+	}
+
+	/**
+	 * Convert a string representing a 24/32bit hex color value into a Color
+	 * object. The following color names are also supported: white, black, red,
+	 * green, blue, orange, yellow, pink, turquoise, gray and none (null).
+	 * Examples of possible hex color values are: #C3D9FF, #6482B9 and #774400,
+	 * but note that you do not include the "#" in the string passed in
+	 * 
+	 * @param colorString
+	 *            the 24/32bit hex string value (ARGB)
+	 * @return java.awt.Color (24bit RGB on JDK 1.1, 24/32bit ARGB on JDK1.2)
+	 * @exception NumberFormatException
+	 *                if the specified string cannot be interpreted as a
+	 *                hexidecimal integer
+	 */
+	public static Color parseColor(String colorString)
+			throws NumberFormatException
+	{
+		if (colorString.equalsIgnoreCase("white"))
+		{
+			return Color.white;
+		}
+		else if (colorString.equalsIgnoreCase("black"))
+		{
+			return Color.black;
+		}
+		else if (colorString.equalsIgnoreCase("red"))
+		{
+			return Color.red;
+		}
+		else if (colorString.equalsIgnoreCase("green"))
+		{
+			return Color.green;
+		}
+		else if (colorString.equalsIgnoreCase("blue"))
+		{
+			return Color.blue;
+		}
+		else if (colorString.equalsIgnoreCase("orange"))
+		{
+			return Color.orange;
+		}
+		else if (colorString.equalsIgnoreCase("yellow"))
+		{
+			return Color.yellow;
+		}
+		else if (colorString.equalsIgnoreCase("pink"))
+		{
+			return Color.pink;
+		}
+		else if (colorString.equalsIgnoreCase("turqoise"))
+		{
+			return new Color(0, 255, 255);
+		}
+		else if (colorString.equalsIgnoreCase("gray"))
+		{
+			return Color.gray;
+		}
+		else if (colorString.equalsIgnoreCase("none"))
+		{
+			return null;
+		}
+
+		int value;
+		try
+		{
+			value = (int) Long.parseLong(colorString, 16);
+		}
+		catch (NumberFormatException nfe)
+		{
+			value = Long.decode(colorString).intValue();
+		}
+
+		return new Color(value);
+	}
+
+	/**
+	 * Returns a hex representation for the given color.
+	 * 
+	 * @param color
+	 *            Color to return the hex string for.
+	 * @return Returns a hex string for the given color.
+	 */
+	public static String getHexColorString(Color color)
+	{
+		return Integer.toHexString((color.getRGB() & 0x00FFFFFF)
+				| (color.getAlpha() << 24));
+	}
+
+	/**
+	 * Reads the given filename into a string.
+	 * 
+	 * @param filename
+	 *            Name of the file to be read.
+	 * @return Returns a string representing the file contents.
+	 * @throws IOException
+	 */
+	public static String readFile(String filename) throws IOException
+	{
+		BufferedReader reader = new BufferedReader(new InputStreamReader(
+				new FileInputStream(filename)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource created by the following constructor: *InputStreamReader*. The resource is referenced at lines 1707, 1710, 1715. The resource is closed at line 1718. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure the object returned by *InputStreamReader()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).